src/nxdoc/nuxeo-server/runtime-and-component-model/runtime-and-component-model-tutorials/how-to-create-a-service.md

```diff
@@ -0,0 +1,417 @@
+      <property name="myProp" value="myValue" />
+      <!-- previous configuration here... -->
+
+    </component>
+    ```
+
+    From the component code, that property can be retrieved from the Runtime context, when activating the component:
+    ```
+    public class MyComponent extends DefaultComponent {
+
+        @Override
+        public void activate(ComponentContext context) {
+            String myPropValue = (String) context.getPropertyValue("myProp");
+        }
+
+    }
+    ```
+1.  Contribute a property to the default service `ConfigurationService`. This property can be defined alongside the component and service declarations. Note that this property value **will be able to be overridden** by another contribution, and will be able to be documented specifically:
+    ```
+    <?xml version="1.0"?>
+    <component name="org.mycompany.myproject.MyService">
+
+      <implementation class="org.mycompany.myproject.MyComponent" />
+      <!-- previous configuration here... -->
+
+      <extension target="org.nuxeo.runtime.ConfigurationService" point="configuration">
+        <documentation>
+          This configuration property can be documented here. HTML tags are accepted.
+        </documentation>
+        <property name="org.mycompany.myproject.myprop">myDefaultValue</property>
+      </extension>
+
+    </component>
+    ```
+
+    From the component code, that property can be retrieved from the Runtime framework, after starting the component, for instance (to make sure ConfigurationService is available):
+    ```
+    import org.nuxeo.runtime.services.config.ConfigurationService;
+
+    public class MyComponent extends DefaultComponent {
+
+        protected static final String MY_PROP_NAME = "org.mycompany.myproject.myprop";
+
+        @Override
+        public void start(ComponentContext context) {
+            super.start(context);
+            String myPropValue = Framework.getService(ConfigurationService.class).getString(MY_PROP_NAME);
+        }
+
+    }
+    ```
+    It can also be retrieved from other Java classes in the code, outside the component implementation.
+1.  Yet another alternative is possible if this configuration property needs to be changed easily depending on the infrastructure, from `nuxeo.conf` properties or [Configuration Templates]({{page page='configuration-templates'}}):
+    ```
+    import org.nuxeo.runtime.services.config.ConfigurationService;
+
+    public class MyComponent extends DefaultComponent {
+
+        protected static final String MY_PROP_NAME = "org.mycompany.myproject.myprop";
+
+        @Override
+        public void start(ComponentContext context) {
+            super.start(context);
+            String myPropValue = Framework.getProperty(MY_PROP_NAME);
+        }
+
+    }
+    ```
+1.  Last but not least, using the `ConfigurationService` and the environment property can be combined, by using a variable in the ConfigurationService contribution:
```

Comment:
This is not a real environment property, at least in the context of OS environment. I think we can talk about Nuxeo properties or nuxeo.conf properties, wdty?

Reply:
I removed occurrences of the misleading "environment" word.